src/type-info.c

```diff
@@ -135,7 +135,19 @@ bool list_all_vectors(r_obj* x) {
   if (r_typeof(x) != R_TYPE_list) {
     r_stop_unexpected_type(r_typeof(x));
   }
-  return r_list_all_of(x, &obj_is_vector);
+
```

Comment:
This decreases code quality a little bit, so add TODO note to switch back to `r_list_all_of()`? Also it'll be easier to handle the memory barrier path for altlists in loop wrappers like `r_list_all_of()`.